src/plugins/core/execute/index.ts

```diff
@@ -4,11 +4,15 @@ import { DisabledInterceptor } from './disabledInterceptor';
 import { LazyVariableInterceptor } from './lazyVariableInterceptor';
 import { LogResponseInterceptor } from './logResponseInterceptor';
 import { RegionScopedVariablesInterceptor } from './regionScopedVariablesInterceptor';
+import { RequestDurationEndInterceptor } from './requestDurationEndInterceptor';
```

Comment:
As noted, an Interceptor would probably be better. However, I am not sure if this will then become the LogResponseInterceptor. This would have in this case already two tasks (trigger logResponse hook, manage ProcessedHttpRegion) and that seems to me also not right.

Reply:
I think I'm a bit confused on the recommendation here. Would you like me to move the duration logic to the `LogResponseInterceptor` adding a beforeLoop to the class to start the performance recording?

If so, I suppose my question is where should I store the start of the performance timing duration in the beforeLoop? I'm currently saving it on the response object but agree that's probably not the best spot for it. Should it be a new top level context variable -- maybe something like context.timings or context.metrics? Or did you want me to initialize the ProcessedHttpRegion in the before, save off the duration start and then modify the processed http region in the afterloop?

Appreciate the clarification.

Reply:
In the context is a options object which can store the metadata, maybe de should use this object. I will contribute to your branch to test some ideas.

Reply:
Using the context.options object would have disadvantages that I overlooked. When using `# @ref` the context is cloned and the options object is passed by reference. But this would falsify the duration stored in the options object. For this reason I now create the ProcessedHttpRegion in the beforeLoop and load it again from the array using the id of the current HttpRegion.